logging.dylan

```diff
@@ -874,7 +874,7 @@ define method parse-formatter-pattern
                   end;
                 "thread" =>
                   method (#rest args)
-                    pad(thread-name(current-thread()));
+                    pad(thread-name(current-thread()) | "???");
```

Comment:
Why not format `current-thread-id()` when there is no name>

Reply:
Brilliant.  Will do.